openvdb/openvdb/tree/LeafNodeBool.h

```diff
@@ -79,13 +79,35 @@ class LeafNode<bool, Log2Dim>
     template<typename OtherValueType>
     explicit LeafNode(const LeafNode<OtherValueType, Log2Dim>& other);
 
-    /// Topology copy constructor
+    /// @brief Deprecated topology copy constructor
+    /// @note  This constructor initialises the bool buffer to the ValueMask
+    ///   states (i.e. value will be true if the active state is on and
+    ///   vice-versa). This is not really a "TopologyCopy" and is therefor
+    ///   deprecated. Use the explicit mask/buffer constructor instead:
+    /// @code
+    ///    // build new leaf node with the mask of 'a', but with the mask of
+    ///    // 'b' as the the new value buffer.
+    ///    const LeafNode a = ... ;
+    ///    const LeafNode b = ... ;
+    ///    const LeafNode copy(a.origin(), /*mask=*/a.getValueMask(),
+    ///      /*buff=*/b.getValueMask());
+    /// @endcode
     template<typename ValueType>
+    OPENVDB_DEPRECATED_MESSAGE("Use LeafNodeBool component constructor.")
     LeafNode(const LeafNode<ValueType, Log2Dim>& other, TopologyCopy);
```

Comment:
As part of this, I've decided to deprecate this constructor too as it's not a TopologyCopy. The non-specialized cases (i.e. float, int, etc) don't have this constructor. LeafNodeMasks do, but it makes sense there. I've introduced a component constructor to provide this behaviour more explicitly